### PR TITLE
Rationalize currency unit to "ZEC"

### PR DESCRIPTION
**Pull request by str4d**
Cherry-picked from the upstream PR bitcoin/bitcoin#6504

Part of #2074

### CODE REVIEW

File: src/amount.h [hunk 1]
```diff
@@ -16,6 +16,8 @@ typedef int64_t CAmount;
 static const CAmount COIN = 100000000;
 static const CAmount CENT = 1000000;
 
+extern const std::string CURRENCY_UNIT;
+
 /** No amount larger than this (in satoshi) is valid.
```
Comment (arcalinea) [inline src/amount.h]: s/satoshi/zatoshi?

File: src/amount.h [hunk 2]
```diff
@@ -16,6 +16,8 @@ typedef int64_t CAmount;
...
  *
  * Note that this constant is *not* the total money supply, which in Bitcoin
```
Comment (arcalinea) [inline src/amount.h]: s/Bitcoin/Zcash

File: src/amount.h (hunk 1 above)
Comment (str4d) [reply src/amount.h]: There are other PRs that handle these changes. This PR solely focuses on the currency unit.

File: src/amount.h (hunk 2 above)
Comment (str4d) [reply src/amount.h]: See above.

File: src/amount.h (hunk 1 above)
Comment (arcalinea) [reply src/amount.h]: ok

**Comment by arcalinea** [comment]
ACK, let's try to get the other PR's changing the names in there too though. At least before 1 yr is up :)

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**